system/lib/wasmfs/wasmfs.cpp

```diff
@@ -76,16 +78,78 @@ __wasi_errno_t __wasi_fd_write(__wasi_fd_t fd,
 
   auto lockedFile = file->locked();
 
-  __wasi_size_t num = 0;
+  off_t offset = lockedOpenFile.position();
   for (size_t i = 0; i < iovs_len; i++) {
     const uint8_t* buf = iovs[i].buf;
-    __wasi_size_t len = iovs[i].buf_len;
+    size_t len = iovs[i].buf_len;
 
-    lockedFile.write(buf, len);
-    num += len;
+    // Check if the sum of the buf_len values overflows a uint64_t (file size).
+    if (__wasmfs_offset_t(offset) + __wasmfs_offset_t(len) <
+        __wasmfs_offset_t(offset)) {
```

Comment:
I think this is where we said we could use a function or a macro to check for overflow?